recursion/recurse.rb

```diff
@@ -1,10 +1,32 @@
 def fact(n)
+  if n == 1 || n == 0
+    return 1
+  end
+  return n * fact(n-1)
 end
 
+# 1 1 2 3 5 8 13
 def fib(n)
+  if n == 1 || n == 2
+    return 1
+  end
+  return fib(n-1) + fib(n-2)
 end
 
+# use length of word for base case
+# throw out two letters each time you recurse
+# identify if word is the same forwards and backwards
 def pal(s)
+  if s.length == 1 || s.length == 0
+    return true
+  end
+  if s[0] == s[-1]
+    s.slice!(0)
+    s.slice!(-1)
+    pal(s)
+  else
```

Comment:
Would be nice to test for this first

return false if s[0] != s[-1]

because typically we test all base cases first, then recursive cases.